WalletWasabi.Gui/Global.cs

```diff
@@ -438,130 +452,138 @@ private void NotifyAndLog(string message, string title, NotificationType notific
 
 		public async Task DisposeAsync()
 		{
-			Logger.LogWarning("Process is exiting.", nameof(Global));
+			// Dispose method may be called just once.
+			if (!_disposeRequested)
+			{
+				_disposeRequested = true;
+				StoppingCts.Cancel();
+			}
+			else
+			{
+				return;
+			}
 
-			try
+			using (await InitializationAsyncLock.LockAsync())
 			{
-				StoppingCts?.Cancel();
+				Logger.LogWarning("Process is exiting.", nameof(Global));
 
-				if (!InitializationStarted)
+				try
 				{
-					return;
-				}
+					Logger.LogDebug("Waiting for initialization to complete.", nameof(Global));
 
-				Logger.LogDebug($"Waiting for initialization to complete.", nameof(Global));
+					try
+					{
```

Comment:
Does `InitializationCompleted` make sense anymore?

Reply:
I think we should remove it now. If it is necessary in the future, we can add it back.